tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java

```diff
@@ -142,48 +136,32 @@ private ApiGatewayException loggingFailureReporter(Exception exception) {
     private PublishingRequestCase createPublishingRequest(PublishingRequestCase publishingRequestCase,
                                                           Publication publication, RequestUtils requestUtils)
         throws ApiGatewayException {
-        var username = new Username(requestUtils.username());
-        return REGISTRATOR_PUBLISHES_METADATA_AND_FILES.equals(publishingRequestCase.getWorkflow())
-                   ? createPublishingRequestAndPublishMetadateWithFiles(publishingRequestCase, publication, username)
-                   : createPublishingRequestForNonCurator(publishingRequestCase, publication, username);
-    }
 
-    private PublishingRequestCase createPublishingRequestAndPublishMetadateWithFiles(
-        PublishingRequestCase publishingRequestCase, Publication publication, Username curator)
-        throws ApiGatewayException {
-        publishPublicationAndFiles(publication);
-        return createAutoApprovedTicketForCurator(publishingRequestCase, publication, curator);
-    }
+        var username = new Username(requestUtils.username());
 
-    private PublishingRequestCase createPublishingRequestForNonCurator(PublishingRequestCase publishingRequestCase,
-                                                                       Publication publication, Username curator)
-        throws ApiGatewayException {
-        if (REGISTRATOR_PUBLISHES_METADATA_AND_FILES.equals(publishingRequestCase.getWorkflow())) {
-            publishPublicationAndFiles(publication);
-            return createAutoApprovedTicket(publishingRequestCase, publication, curator);
-        }
-        if (REGISTRATOR_PUBLISHES_METADATA_ONLY.equals(publishingRequestCase.getWorkflow())) {
-            publishMetadata(publication);
-            return createAutoApprovedTicketWhenPublicationContainsMetadataOnly(publishingRequestCase, publication,
-                                                                               curator);
-        } else {
-            return (PublishingRequestCase) publishingRequestCase.persistNewTicket(ticketService);
-        }
```

Comment:
If you take a look at AcceptedPublishingRequestEventHandler, it already does same operations but based on consumed event:

```
    private void handleCompletedPublishingRequest(PublishingRequestCase publishingRequestCase) {
        var publication = fetchPublication(publishingRequestCase.getResourceIdentifier());
        var publishingRequest = fetchPublishingRequest(publishingRequestCase);
        var updatedPublication = toPublicationWithPublishedFiles(publication, publishingRequest);
        var ticketIdentifier = publishingRequest.getIdentifier();
        switch (publishingRequestCase.getWorkflow()) {
            case REGISTRATOR_PUBLISHES_METADATA_ONLY, REGISTRATOR_PUBLISHES_METADATA_AND_FILES -> {
                publishFiles(updatedPublication);
                logger.info(PUBLISHING_FILES_MESSAGE, publication.getIdentifier(), ticketIdentifier);
            }
            case REGISTRATOR_REQUIRES_APPROVAL_FOR_METADATA_AND_FILES -> {
                publishFiles(updatedPublication);
                publishPublication(publication);
                logger.info(PUBLISHING_METADATA_AND_FILES_MESSAGE, publication.getIdentifier(), ticketIdentifier);
            }
            default -> logger.error(UNKNOWN_WORKFLOW_MESSAGE, publishingRequestCase.getWorkflow());
        }
        createDoiRequestIfNeeded(updatedPublication);
    }
```

We remove duplicated code and decrease complexity of handling actions on publication.